src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -1442,6 +1442,9 @@ int64_t GCToOSInterface::QueryPerformanceFrequency()
 //  Time stamp in milliseconds
 uint32_t GCToOSInterface::GetLowPrecisionTimeStamp()
 {
+#if HAVE_CLOCK_GETTIME_NSEC_NP
```

Comment:
If we are fixing this, we should use CLOCK_MONOTONIC_COARSE when available.

It would make sense to move NativeAOT implementation to here and NativeAOT can call this too.
https://github.com/dotnet/runtime/blob/0bce5f69ee027fd148a8d878af5642726ad36a19/src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp#L656

Monotonic counter would be cheaper and will not cause surprises when daylight savings move the clock back and forth.  

I am not sure if fallback to `gettimeofday` is useful. It does not make much difference though, since all platforms we run on have monotonic clock or clock_gettime_nsec_np

Reply:
Sounds reasonable. I'll give it a try tomorrow.

Reply:
Considering that QPC counter does not fallback to `gettimeofday` and just fails to compile, I think this counter should just do the same.

Reply:
I cannot call `GCToOSInterface::GetLowPrecisionTimeStamp()` from `PalGetTickCount64` because it has 32-bit return value. I can change it, of course, but not sure if that's desirable...